In [114]:
import pandas as pd
import sqlite3

In [51]:
df1 = pd.read_csv('data/bom.movie_gross.csv')

In [52]:
df1

title      studio  domestic_gross  \
0                                     Toy Story 3          BV 415000000.00000   
1                      Alice in Wonderland (2010)          BV 334200000.00000   
2     Harry Potter and the Deathly Hallows Part 1          WB 296000000.00000   
3                                       Inception          WB 292600000.00000   
4                             Shrek Forever After        P/DW 238700000.00000   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.      6200.00000   
3383                  Edward II (2018 re-release)          FM      4800.00000   
3384                                     El Pacto        Sony      2500.00000   
3385                                     The Swan  Synergetic      2400.00000   
3386                            An Actor Prepares       Grav.      1700.00000   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]

In [53]:
df1.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [54]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [76]:
df_clean1 = df1.copy()
df_clean1 = df_clean1.drop(['foreign_gross'], axis = 1)
df_clean1.head()

title studio  domestic_gross  year
0                                  Toy Story 3     BV 415000000.00000  2010
1                   Alice in Wonderland (2010)     BV 334200000.00000  2010
2  Harry Potter and the Deathly Hallows Part 1     WB 296000000.00000  2010
3                                    Inception     WB 292600000.00000  2010
4                          Shrek Forever After   P/DW 238700000.00000  2010

In [77]:
pd.set_option('display.float_format', lambda x: '%.5f' % x)

In [78]:
df_clean1.describe()

domestic_gross       year
count      3359.00000 3387.00000
mean   28745845.06698 2013.95807
std    66982498.23736    2.47814
min         100.00000 2010.00000
25%      120000.00000 2012.00000
50%     1400000.00000 2014.00000
75%    27900000.00000 2016.00000
max   936700000.00000 2018.00000

In [79]:
df_clean1['domestic_gross'] = df_clean1['domestic_gross'].median()

In [80]:
df_clean1.isna().sum()

title             0
studio            5
domestic_gross    0
year              0
dtype: int64

In [86]:
df_clean1.dropna(subset=['studio'], inplace=True)

In [87]:
df_clean1.isna().sum()

title             0
studio            0
domestic_gross    0
year              0
dtype: int64

Dropping foreign_gross revenue column due to 1350 values that are missing. It also was imported as an object. We also removed 5 rows for studio to get rid of the remaining null values.

In [88]:
df2 = pd.read_csv('data/tmdb.movies.csv')
df2.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1    33.53300   2010-11-19   
1                      How to Train Your Dragon    28.73400   2010-03-26   
2                                    Iron Man 2    28.51500   2010-05-07   
3                                     Toy Story    28.00500   1995-11-22   
4                                     Inception    27.92000   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1       7.70000       10788  
1                      How to Train Your Dragon       7.70000        7610  
2                                    Iron Man 2       6.80000       12368  
3                                     Toy Story       7.90000       10174  
4                                     Inception       8.30000       22186

We are putting the tmdb.movies.csv on ice for now due to the only column we find value in is the genre ids. Until we can link those to an actual genre instead of just an ID. We are putting this one on hold.

In [96]:
df3 = pd.read_csv('data/tn.movie_budgets.csv', parse_dates=['release_date'])
df3.head()

id release_date                                        movie  \
0   1   2009-12-18                                       Avatar   
1   2   2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2   3   2019-06-07                                 Dark Phoenix   
3   4   2015-05-01                      Avengers: Age of Ultron   
4   5   2017-12-15            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [90]:
df3.isna().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

In [93]:
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [95]:
df3['movie'].duplicated().sum()

84

Release date to date/time and change the strings to ints that need to change. There are duplicates that need to be removed as well.

In [98]:
df4 = pd.read_table('data/rt.movie_info.tsv')
df4

id                                           synopsis rating  \
0        1  This gritty, fast-paced, and innovative police...      R   
1        3  New York City, not-too-distant-future: Eric Pa...      R   
2        5  Illeana Douglas delivers a superb performance ...      R   
3        6  Michael Douglas runs afoul of a treacherous su...      R   
4        7                                                NaN     NR   
...    ...                                                ...    ...   
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1557  1998  Based on a novel by Richard Powell, when the l...      G   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
0                   Action and Adventure|Classics|Drama    William Friedkin   
1                     Drama|Science Fiction and Fantasy    David Cronenberg   
2                     Drama|Musical and Performing Arts      Allison Anders   
3                            Drama|Mystery and Suspense      Barry Levinson   
4                                         Drama|Romance      Rodney Bennett   
...                                                 ...                 ...   
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
0                                       Ernest Tidyman   Oct 9, 1971   
1                         David Cronenberg|Don DeLillo  Aug 17, 2012   
2                                       Allison Anders  Sep 13, 1996   
3                      Paul Attanasio|Michael Crichton   Dec 9, 1994   
4                                         Giles Cooper           NaN   
...                                                ...           ...   
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
0     Sep 25, 2001      NaN         NaN  104 minutes                NaN  
1      Jan 1, 2013        $     600,000  108 minutes  Entertainment One  
2     Apr 18, 2000      NaN         NaN  116 minutes                NaN  
3     Aug 27, 1997      NaN         NaN  128 minutes                NaN  
4              NaN      NaN         NaN  200 minutes                NaN  
...            ...      ...         ...          ...                ...  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures  

[1560 rows x 12 columns]

In [99]:
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


RT Movie Reviews:
Need to connect movie names to each column before we can even talk about cleaning this one.

In [102]:
df5 = pd.read_table('data/rt.reviews.tsv', encoding='latin1')
df5.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [103]:
df5.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


rt.review.tsv: Unless we can connect something on this table to the movies, we can't use any of this information.

In [115]:
conn = sqlite3.connect('data/im.db')
cursor = conn.cursor()

In [116]:
df6 = pd.read_sql("""
SELECT *
FROM 
    sqlite_master
""", conn)
df6.head(10)

type           name       tbl_name  rootpage  \
0  table   movie_basics   movie_basics         2   
1  table      directors      directors         3   
2  table      known_for      known_for         4   
3  table     movie_akas     movie_akas         5   
4  table  movie_ratings  movie_ratings         6   
5  table        persons        persons         7   
6  table     principals     principals         8   
7  table        writers        writers         9   

                                                 sql  
0  CREATE TABLE "movie_basics" (\n"movie_id" TEXT...  
1  CREATE TABLE "directors" (\n"movie_id" TEXT,\n...  
2  CREATE TABLE "known_for" (\n"person_id" TEXT,\...  
3  CREATE TABLE "movie_akas" (\n"movie_id" TEXT,\...  
4  CREATE TABLE "movie_ratings" (\n"movie_id" TEX...  
5  CREATE TABLE "persons" (\n"person_id" TEXT,\n ...  
6  CREATE TABLE "principals" (\n"movie_id" TEXT,\...  
7  CREATE TABLE "writers" (\n"movie_id" TEXT,\n  ...

In [117]:
df6 = pd.read_sql("""
SELECT *
FROM 
    movie_basics
""", conn)
df6.head(10)

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   
5  tt0111414                      A Thin Life                 A Thin Life   
6  tt0112502                          Bigfoot                     Bigfoot   
7  tt0137204                  Joe Finds Grace             Joe Finds Grace   
8  tt0139613                       O Silêncio                  O Silêncio   
9  tt0144449            Nema aviona za Zagreb       Nema aviona za Zagreb   

   start_year  runtime_minutes                      genres  
0        2013        175.00000          Action,Crime,Drama  
1        2019        114.00000             Biography,Drama  
2        2018        122.00000                       Drama  
3        2018              NaN                Comedy,Drama  
4        2017         80.00000        Comedy,Drama,Fantasy  
5        2018         75.00000                      Comedy  
6        2017              NaN             Horror,Thriller  
7        2017         83.00000  Adventure,Animation,Comedy  
8        2012              NaN         Documentary,History  
9        2012         82.00000                   Biography

In [118]:
df6 = pd.read_sql("""
SELECT *
FROM 
    movie_akas
""", conn)
df6.head(10)

movie_id  ordering                                    title region  \
0  tt0369610        10                            Джурасик свят     BG   
1  tt0369610        11                        Jurashikku warudo     JP   
2  tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3  tt0369610        13                  O Mundo dos Dinossauros     BR   
4  tt0369610        14                           Jurassic World     FR   
5  tt0369610        15                           Jurassic World     GR   
6  tt0369610        16                           Jurassic World     IT   
7  tt0369610        17                            Jurski svijet     HR   
8  tt0369610        18                             Olam ha'Yura     IL   
9  tt0369610        19           Jurassic World: Mundo Jurásico     MX   

  language        types   attributes  is_original_title  
0       bg         None         None            0.00000  
1     None  imdbDisplay         None            0.00000  
2     None  imdbDisplay         None            0.00000  
3     None         None  short title            0.00000  
4     None  imdbDisplay         None            0.00000  
5     None  imdbDisplay         None            0.00000  
6     None  imdbDisplay         None            0.00000  
7     None  imdbDisplay         None            0.00000  
8       he  imdbDisplay         None            0.00000  
9     None  imdbDisplay         None            0.00000

In [142]:
df6 = pd.read_sql("""
SELECT *
FROM 
    persons
""", conn)
df6.head(10)

person_id       primary_name  birth_year  death_year  \
0  nm0061671  Mary Ellen Bauder         NaN         NaN   
1  nm0061865       Joseph Bauer         NaN         NaN   
2  nm0062070         Bruce Baum         NaN         NaN   
3  nm0062195       Axel Baumann         NaN         NaN   
4  nm0062798        Pete Baxter         NaN         NaN   
5  nm0062879     Ruel S. Bayani         NaN         NaN   
6  nm0063198              Bayou         NaN         NaN   
7  nm0063432      Stevie Be-Zet         NaN         NaN   
8  nm0063618          Jeff Beal  1963.00000         NaN   
9  nm0063750    Lindsay Beamish         NaN         NaN   

                                 primary_profession  
0         miscellaneous,production_manager,producer  
1        composer,music_department,sound_department  
2                        miscellaneous,actor,writer  
3  camera_department,cinematographer,art_department  
4  production_designer,art_department,set_decorator  
5         director,production_manager,miscellaneous  
6                                             actor  
7                               composer,soundtrack  
8              composer,music_department,soundtrack  
9                             actress,miscellaneous

In [143]:
df6.isna().sum()

person_id                  0
primary_name               0
birth_year            523912
death_year            599865
primary_profession     51340
dtype: int64

Include: movie_basics, movie_ratings, (connected through principals to directors and writers being identified by person.) The parenthesis are the extra part of things.

In [123]:
!pip install pandasql

  Created wheel for pandasql: filename=pandasql-0.7.3-py3-none-any.whl size=26784 sha256=82a35cce026bfe4b387087cf29fbc00594c12640212622412923aa11530bbf87
  Stored in directory: c:\users\mackoy\appdata\local\pip\cache\wheels\63\e8\ec\75b1df467ecf57b6ececb32cb16f4e86697cbfe55cb0c51f07
Successfully built pandasql


In [132]:
movie_joined = pd.read_sql("""
SELECT 
    primary_title, 
    start_year, 
    runtime_minutes, 
    genres,
    averagerating, 
    numvotes
FROM movie_basics
JOIN movie_ratings
    USING(movie_id)
""", conn)
movie_joined.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   primary_title    73856 non-null  object 
 1   start_year       73856 non-null  int64  
 2   runtime_minutes  66236 non-null  float64
 3   genres           73052 non-null  object 
 4   averagerating    73856 non-null  float64
 5   numvotes         73856 non-null  int64  
dtypes: float64(2), int64(2), object(2)
memory usage: 3.4+ MB


In [137]:
movie_joined.head()

primary_title  start_year  runtime_minutes  \
0                        Sunghursh        2013        175.00000   
1  One Day Before the Rainy Season        2019        114.00000   
2       The Other Side of the Wind        2018        122.00000   
3                  Sabse Bada Sukh        2018              NaN   
4         The Wandering Soap Opera        2017         80.00000   

                 genres  averagerating  numvotes  
0    Action,Crime,Drama        7.00000        77  
1       Biography,Drama        7.20000        43  
2                 Drama        6.90000      4517  
3          Comedy,Drama        6.10000        13  
4  Comedy,Drama,Fantasy        6.50000       119

In [165]:
profession_joined = pd.read_sql("""
SELECT
    primary_name,
    movie_id,
    category
FROM
    persons
JOIN
    principals
    USING(person_id)
    
""", conn)

In [156]:
profession_joined.head()

primary_name   movie_id  category
0  Mary Ellen Bauder  tt2398241  producer
1       Joseph Bauer  tt0433397  composer
2       Joseph Bauer  tt1681372  composer
3       Joseph Bauer  tt2281215  composer
4       Joseph Bauer  tt2387710  composer

In [159]:
persons_joined.value_counts()

primary_name          movie_id    category
Ezra Brown            tt6239106   composer    3
Pavel Pavlov          tt6144460   producer    3
Jake Baker            tt6017666   actor       3
Riitta Jalonen        tt10381796  director    2
István Bende          tt7543948   actor       2
                                             ..
Goran Bregovic        tt7180386   actor       1
Goran Bukovcan        tt8925602   writer      1
Goran Cace            tt3833120   editor      1
                      tt4150188   editor      1
Þórður Aðalbjörnsson  tt2343384   actor       1
Length: 1026893, dtype: int64

In [161]:
movie_profession = persons_joined.merge(movie_joined, 
                                        on ='movie_id', 
                                        how = 'inner')

KeyError: 'movie_id'